File: src/main/scala/org/nygenome/als/graphdb/value/Pathway.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord

case class Pathway(uniprotId:String, reactomeId: String, eventName:String, evidenceCode:String,
                   species:String, id:String) {
}

object Pathway extends ValueTrait {
  val pathwayHeadings:String = "UniProt_ID\tReactome_ID\tURL\tEvent_Name\tEvidence_Code\tSpecies"
  val reactomeBaseUrl:String ="https://reactome.org/PathwayBrowser/#/"

  def parseCSVRecord(record: CSVRecord): Pathway = {
      new Pathway(
        record.get("UniProt_ID"),
        record.get("Reactome_ID"),
        record.get("Event_Name"),
        record.get("Evidence_Code"),
        record.get("Species"),
        record.get("UniProt_ID") +":" +record.get("Reactome_ID")
      )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/Pathway.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

/**
 * Created by fcriscuo on 2/5/20.
 */
data class Pathway(
        val uniprotId: String, val reactomeId: String, val eventName: String, val evidenceCode: String,
        val species: String, val id: String
) {
    val pathwayHeadings: String = "UniProt_ID\tReactome_ID\tURL\tEvent_Name\tEvidence_Code\tSpecies"
    val reactomeBaseUrl: String = "https://reactome.org/PathwayBrowser/#/"

    fun isHuman(): Boolean = species.equals("Homo sapiens",ignoreCase = true)

    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): Pathway =
                Pathway(record.get("UniProt_ID"),
                        record.get("Reactome_ID"),
                        record.get("Event_Name"),
                        record.get("Evidence_Code"),
                        record.get("Species"),
                        // composite identifier
                        record.get("UniProt_ID") + ":" + record.get("Reactome_ID")
                )
    }
}